### PR TITLE
Update Developer Guide and documentation

### PR DESCRIPTION
**Pull request by L1uY1jun**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -176,7 +176,7 @@ An applicant can have at most 2 majors (including 0).
 </div>
 
 <div markdown="span" class="alert alert-primary">:bulb: **Tip:**
-Application status is **case-insensitive**, entering the status PENDING, ACCEPTED or REJECTED in any combination of upper or lower case is acceptable.
+Application status is **case-insensitive**, entering the status PENDING, ACCEPTED or REJECTED in any combination of upper or lower case is acceptable. This applies to the other features who make use of application status in our guide
```
Comment (benjytan45678) [inline docs/UserGuide.md]: other features ***which***

File: src/main/java/seedu/trackascholar/ui/AlertWindow.java [hunk 2]
```diff
@@ -34,6 +34,7 @@ public boolean display(String message) {
                 .ifPresent(response ->
                         setIsConfirmed());
         return isConfirmed;
+        //@@author
```
Comment (benjytan45678) [inline src/main/java/seedu/trackascholar/ui/AlertWindow.java:37]: This line is missing sth for @author?

File: docs/UserGuide.md (hunk 1 above)
Comment (L1uY1jun) [reply docs/UserGuide.md]: Noted, changes made

**Comment by benjytan45678** [comment]
LGTM, merged.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -445,66 +569,66 @@ The following activity diagram summarizes what happens when a user executes a un
...
+| `* * *` | administrator (new user)                             | see usage instructions easily                                               | discover functionalities provided by the application                                        |
+| `* * *` | administrator (new user)                             | operate with clear and concise commands                                     | learn easily and grow proficiency with time                                                 |
+| `* * *` | administrator                                        | retrieve previously stored application data                                 | access stored database easily                                                               |
+| `* * *` | administrator                                        | save fresh data easily                                                      | avoid losing all my data                                                                    |
+| `* * *` | administrator                                        | add a new applicant to my database                                          | keep track of organize all the scholarship applications                                     |
```
Comment (benjytan45678) [inline docs/DeveloperGuide.md:611]: Typo in keep track of ***organize***?

Comment (benjytan45678) [reply docs/DeveloperGuide.md:611]: I reckon we can change database to ***application***

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -445,66 +569,66 @@ The following activity diagram summarizes what happens when a user executes a un
...
+| `* * *` | administrator                                        | delete an applicant from my database                                        | remove data of scholarship applications that is no longer needed                            |
+| `* * *` | administrator                                        | edit the details of an applicant in my database                             | keep track of their most updated details                                                    |
+| `* * *` | administrator managing many scholarship applications | view the list of all applicants in my database                              | have an overview of all scholarship applicants in the list of applications                  |
```
Comment (benjytan45678) [inline docs/DeveloperGuide.md:614]: the list of applications to ***one go***?

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -445,66 +569,66 @@ The following activity diagram summarizes what happens when a user executes a un
...
+| `* * *` | administrator managing many scholarship applications | find applicants by their name                                               | retrieve details of an applicant without having to go through the entire list               |
+| `* * *` | administrator managing many scholarship applications | find applicants by the scholarship they applied for                         | retrieve details of all applicants who applied for the same scholarships                    |
```
Comment (benjytan45678) [inline docs/DeveloperGuide.md:616]: applied for the same ***scholarship***

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (L1uY1jun) [reply docs/DeveloperGuide.md:611]: Noted, changes made

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (L1uY1jun) [reply docs/DeveloperGuide.md:614]: Noted, changes made

**Pull request merged**